### ISSUE TITLE
Allow trace/span propagation exporting

### ISSUE THREAD

**Issue by phillipuniverse** [DataDog/dd-trace-cpp#55]
I am just getting started on integrating this project for otel propagation (coming from the opentelemetry-cpp-config project which is too much in limbo). It looks like this has a similar issue in the default otel nginx instrumentation where I cannot enable tracing (extraction, injection, log messages) without also having the Datadog agent up to receive traces.

My use case is running locally, I want to be able to test trace propagation without exporting any of the spans that are generated from nginx.

I didn't see how this was possible in the current config, is it?

**Comment by dgoffredo** [comment]
Hi, Phillip.

> My use case is running locally, I want to be able to test trace propagation without exporting any of the spans that are generated from nginx.

Are you using the [nginx-datadog][1] module, which uses this library? If so, you can set the environment variable [DD_TRACE_ENABLED=false][2]. This will [prevent][3] the tracer from attempting to contact the Datadog agent.

Regardless of whether you're using that nginx module, within this library the configuration option [bool TracerConfig::report_traces][4] controls this behavior. `report_traces` is overridden by the `DD_TRACE_ENABLED` environment variable if the environment variable is present during configuration.

[1]: https://github.com/DataDog/nginx-datadog
[2]: https://github.com/DataDog/dd-trace-cpp/blob/e9b715d1c66f9dc4ccb3a0561475336fe92431af/src/datadog/tracer_config.cpp#L309-L312
[3]: https://github.com/DataDog/dd-trace-cpp/blob/e9b715d1c66f9dc4ccb3a0561475336fe92431af/src/datadog/tracer_config.cpp#L314-L316
[4]: https://github.com/DataDog/dd-trace-cpp/blob/e9b715d1c66f9dc4ccb3a0561475336fe92431af/src/datadog/tracer_config.h#L45-L50

**Comment by phillipuniverse** [comment]
Sorry about that, I did indeed accidentally land in this project (when I was looking at environment variables) and intended to open this issue in the nginx-datadog project.

I'm actually going to close this and re-open there. @dgoffredo great insight, but my understanding is that DD_TRACE_ENABLED will turn off all propagation.

I opened an issue at https://github.com/DataDog/nginx-datadog/issues/39
